examples/3d/bloom.rs

```diff
@@ -0,0 +1,168 @@
+        emissive: Color::rgb_linear(5.2, 1.2, 0.8), // 3. Set StandardMaterial::emissive using Color::rgb_linear, for entities we want to apply bloom to
+        ..Default::default()
+    });
+    let material_non_emissive = materials.add(StandardMaterial {
+        ..Default::default()
+    });
+
+    let mesh = meshes.add(
+        shape::Icosphere {
+            radius: 0.5,
+            subdivisions: 5,
+        }
+        .into(),
+    );
+
+    for x in -10..10 {
+        for z in -10..10 {
+            let mut hasher = DefaultHasher::new();
+            (x, z).hash(&mut hasher);
+            let rand = hasher.finish() % 2 == 0;
+
+            let material = if rand {
+                material.clone()
+            } else {
+                material_non_emissive.clone()
+            };
+
+            commands.spawn((
+                PbrBundle {
+                    mesh: mesh.clone(),
+                    material,
+                    transform: Transform::from_xyz(x as f32 * 2.0, 0.0, z as f32 * 2.0),
+                    ..Default::default()
+                },
+                Bouncing,
+            ));
+        }
+    }
+}
+
+// ------------------------------------------------------------------------------------------------
+
+fn setup_instructions(mut commands: Commands, asset_server: Res<AssetServer>) {
+    // UI camera
+    commands.spawn(Camera2dBundle {
+        camera: Camera {
+            priority: -1,
+            ..default()
+        },
+        ..default()
+    });
+
+    commands.spawn(
+        TextBundle::from_section(
+            "",
+            TextStyle {
+                font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                font_size: 12.0,
+                color: Color::WHITE,
+            },
+        )
+        .with_style(Style {
+            position_type: PositionType::Absolute,
+            position: UiRect {
+                top: Val::Px(10.0),
+                left: Val::Px(10.0),
+                ..default()
+            },
+            ..default()
+        }),
+    );
+}
+
+fn update_bloom_settings(
+    mut camera: Query<&mut BloomSettings>,
+    mut text: Query<&mut Text>,
+    keycode: Res<Input<KeyCode>>,
+) {
+    let mut bloom_settings = camera.single_mut();
+    let mut text = text.single_mut();
+    let text = &mut text.sections[0].value;
+
+    *text = "BloomSettings:".to_string();
+    text.push_str(&format!("Threshold: {}\n", bloom_settings.threshold));
+    text.push_str(&format!("Knee: {}\n", bloom_settings.knee));
+    text.push_str(&format!("Scale: {}\n", bloom_settings.scale));
+    text.push_str(&format!("Intensity: {}\n", bloom_settings.intensity));
+
+    text.push_str("\n");
+
+    text.push_str("Controls (-/+):\n");
+    text.push_str("Q/W - Threshold\n");
+    text.push_str("E/R - Knee\n");
+    text.push_str("A/S - Scale\n");
+    text.push_str("D/F - Intensity\n");
+
+    if keycode.pressed(KeyCode::Q) {
+        bloom_settings.threshold -= 1.0;
```

Comment:
All of these settings should be scaled with `time.delta_seconds()` if we're going to update them every frame a key is pressed.